### PR TITLE
migrate to docker compose v2 in integration tests

### PR DESCRIPTION
**Pull request by paulcacheux**
### Motivation

#incident-27281

### CODE REVIEW

File: test/integration/utils/compose.go [hunk 1]
```diff
@@ -98,14 +98,14 @@ func (c *ComposeConf) Stop() ([]byte, error) {
 	if c.RemoveRebuildImages {
 		args = append(args, "--rmi", "all")
 	}
-	runCmd := exec.Command("docker-compose", args...)
+	runCmd := exec.Command("compos", args...)
```
Comment (clamoriniere) [inline test/integration/utils/compose.go:101]: I think we have a typo here

```suggestion
	runCmd := exec.Command("compose", args...)
```

File: test/integration/utils/compose.go [hunk 2]
```diff
@@ -98,14 +98,14 @@ func (c *ComposeConf) Stop() ([]byte, error) {
...
 	return runCmd.CombinedOutput()
 }
 
 // ListContainers lists the running container IDs
 func (c *ComposeConf) ListContainers() ([]string, error) {
 	runCmd := exec.Command(
-		"docker-compose",
+		"compos",
```
Comment (clamoriniere) [inline test/integration/utils/compose.go:108]: Same typo

```suggestion
		"compose",
```

File: test/integration/utils/compose.go (hunk 1 above)
Comment (spencergilbert) [reply test/integration/utils/compose.go]: 😭 that would definitely help. Thanks!

**Review by sblumenthal — APPROVED** [review]

**Comment by ofek** [comment]
Can we use Docker directly? See:

```
❯ docker run --rm docker:26.1-cli /usr/local/bin/docker compose version
Docker Compose version v2.27.0
```

**Comment by spencergilbert** [comment]
> Can we use Docker directly? See:
> 
> ```
> ❯ docker run --rm docker:26.1-cli /usr/local/bin/docker compose version
> Docker Compose version v2.27.0
> ```

That had failed to work at some point in testing yesterday, we could revisit though.

**Comment by ofek** [comment]
Do you recall what didn't work?

**Comment by spencergilbert** [comment]
> Do you recall what didn't work?

IIRC those are alpine based, and there were issues with cgo?

**Comment by ofek** [comment]
Oh, perhaps I misread, are we not invoking the binaries as a subprocess?

**Review by ofek — APPROVED** [review]

**Pull request merged**